articles/intro-to-C++-part01/index.md

```diff
@@ -0,0 +1,145 @@
+
+- Line 2 - **#include<iostream>**
+
+C++, by itself, cannot do input or output, so we add functionality to C++ by including libraries. For example, :<iostream> is a library that has commands for input from the keyboard and output to the screen.
+
+- Line 3 - **using namespace std;**
+
+It is something that you cannot avoid and have to write in every single program that you will write in your whole life. As defined by Danoja Das on [medium](https://medium.com/breaktheloop/why-using-namespace-std-is-used-after-including-iostream-dc5ae45db652), when we run a program to print something, “using namespace std” says if you find something that is not declared in the current scope go and check std. using namespace std; are used. It is because computer needs to know the code for the cout, cin functionalities and it needs to know which namespace are defined.
+
+- Line 4 - **int main ( ) {**
+
+The aforementioned line tells the compiler that this is the beginning of the main program. **{** means the start of the block of the program.
+
+- Line 5 - **cout<<"Hello World"<<endl;**
+
+**cout** is our "output to the screen" statement. **"endl"** takes you the next line. **";"** ends a C++ statement.
+
+- Line 6 - **return 0**
+
+This means that this is the end of our program.
+
+- Line 7 - **}**
+
+The line above is our closing bracket to end the program.
+
+Congratulations! You just wrote your first C++ program.
+
+Let's write another program that would read two numbers and output the sum of them.
+`  
+
+     // program to add two numbers
+
+     #include <iostream>
+     using namespace std;
+     int main(){
+     cout<<"enter your first number to add"<<endl;
+     int num_one;
+     cin>>num_one;
+     cout<<" Enter your second number to add"<<endl;
+     int num_two;
+    cin>>num_two;
+    int sum;
+    sum = num_one + num_two;
+    cout<<num_one<<"+"<<num_two<<"="<<sum<<endl;
+      return0;
+    }`  
+
+Here's what the output would like to this program:
+
+         enter your first number to add: 7
+
+         enter your second number to add: 13
+
+         7+13 = 20
+
+Let's discuss the things that are new to us in this program above.
+
+1. **int num_one** - This means that we are reserving space in memory to hold an integer (int). And we are naming that space num_one.
+
+Integers hold positive as well as negative values (and 0) and include non-decimal numbers up to about +-2 billion, which gives you an exact value between +- 2 billion.
+
+Just like int, we also have double. It can hold large as well as small numbers. It can also be an exponent. Here is how we use it in a program.
+
+`double first_one;`   
+
+2. **cin>>num_one** - This is the way of accepting an input from the user in the same memory location that we reserved earlier.
+
+And that is all you need to know to write basic programs in C++. We will go over other new things as we continue.
```

Comment:
Are you sure that's 'all' they need to know, or is it better to state this as having provided some of the basic building blocks to get started?